.github/workflows/marine-tests.yml

```diff
@@ -0,0 +1,96 @@
+name: "marine-tests"
+
+on:
+  pull_request:
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  air:
+    name: "Run marine tests"
+    runs-on: builder
+
+    env:
+      RUST_BACKTRACE: 1
+      CARGO_TERM_COLOR: always
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly-2022-06-29
+          target: wasm32-wasi
+          components: rustfmt, clippy
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: "Cache rust"
+        uses: Swatinem/rust-cache@v1
+
+      - name: "Install wasm-pack"
+        run: cargo install wasm-pack
+
+      - name: "Install marine cli"
+        run: cargo install --path tools/cli
+
+      - name: "cargo fmt"
+        run: cargo fmt --all -- --check
+
+      - name: "Build marine-js"
+        run: ./build.sh
+        working-directory: marine-js
+
+      - name: "Install marine-js npm dependencies"
+        uses: bahmutov/npm-install@v1
+        with:
+          working-directory: marine-js/npm-package
+
+      - name: "Build marine-js npm package"
+        run: npm run build
+        working-directory: marine-js/npm-package
+
+      - name: "Build exampes"
+        run: ./build.sh
+        working-directory: examples
+
+      - name: "Build marine wasm-tests"
+        run: ./build.sh
+        working-directory: marine/tests/wasm_tests
+
+      - name: "Build core wasm-tests"
+        run: ./build.sh
+        working-directory: core/tests/wasm_tests
+
+      - name: "cargo check"
+        run: |
+          cargo check -v --all-features \
+            -p marine-core \
+            -p marine \
+            -p fluence-app-service \
+            -p marine \
+            -p mrepl
+
+      - name: "cargo test"
+        run: |
+          cargo test --release -v --all-features \
+            -p marine-core \
+            -p marine \
+            -p fluence-app-service \
+            -p marine \
+            -p mrepl \
+
+      - name: "cargo clippy"
+        run: cargo clippy -v
+
+      - name: "npm test marine-js"
+        run: npm run test
+        working-directory: marine-js/npm-package
+
+      - name: "Setup tmate"
```

Comment:
I think it's better to enable this via manual `workflow_dispatch`. Tests will fail on some branches, I don't think it's an enough requirement to keep instance running waiting for ssh.